### ISSUE TITLE
✨ Add computed Gov Employee data to User model (computed based on experience)

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#12217]
## ✨ Feature

We're adding the collection of government information to experiences. This makes a lot of the info on the Government Employee section of the profile redundant. Instead of asking users about it twice (once in Experiences once in Profile) we'll ask it only in work experiences, and when any work experiences change we'll re-compute centralized values which are stored with the user model, making analytics and display easier.

## 🕵️ Details

In scope for this ticket:
- adding and renaming fields on the users table
- compute and update gov_employee fields on users table when a 'current' work experience is saved or updated
- updated graphql schema so fields in User type continue to point to same underlying fields

Out of scope for this ticket:
- modifying UI or API

## 🎨 Design

@Jerryescandon 

## 📸 Screenshot

<details><summary>📸 User profile  </summary>
<p>

[Figma file](https://www.figma.com/design/vZEg5z6aXiJPNB07LamrgY/Admin---Users?node-id=270-26057&t=lufTPLHrZd9MZYqo-1) 

![image](https://github.com/user-attachments/assets/2a0621d3-abcf-431d-9208-cab29b8fb80f)

</p>
</details>

## 🙋‍♀️ Proposed Implementation

Rename existing fields in users table to mark them as computed   
  - is_gov_employee -> computed_is_gov_employee
  - gov_employee_type -> computed_gov_employee_type
  - current_classification -> computed_current_classification
  - department -> computed_department

Create new fields in users table   
  - computed_gov_position_type
  - computed_gov_end_date

Create a new method in User.php, `computeGovEmployeeProfile` (or `computeGovEmployeeProfileData`? Or something else?), similar to `computeFinalDecision` is PoolCandidate.php. This function _may_ update the new computed fields:
```
**current group & level selection logic**

1. First show the Group & Level from whichever government experience is marked as current
   1. Except if "employment type = contractor" or "employment type = student". These will be ignored. No group & level are collected for them anyway.
2. If multiple are marked current, show the one with the most recent start date
3. If multiple are marked current and have the same start date, prioritize the experience based on the employment type (though this probably should never happen):
   1. Show indeterminate first
   2. If not indeterminate, show Term
   3. Last show casual
4. If multiple are marked current, have the same start date and the same "employment type", then select based on the position type. (This actually is possible, for example you start a new substantive but are immediately put into an acting.)
   1. Show acting first
   2. If no acting, show secondment
   3. If no acting or secondment, show assignment
   4. Last show substantive
5. And if for some reason multiple experiences are marked current, have the same start date, have the same employment type and the same position type, then just pick whichever was created most recently by the user. (this should also never happen though)

``` 

 - Create a laravel Event, `ExperienceSaved`, which dispatches [whenever a WorkExperience is saved](https://laravel-news.com/model-events). See AssessmentResultSaved as an example.
- Create a Laravel App Listener, ComputeGovEmployeeProfileData (open to other names). Similarly to how ComputeCandidateAssessmentStatus is set up to run in response to an AssessmentResultSaved event, set up the new listener to run in response to an ExperienceSaved event.
    - The ComputeGovEmployeeProfileData listener runs user->computeGovEmployeeProfile in the following cases:
        - The work experience is current (ie has no end date)
        - This is an update event, and before the update the experience _was_ current, (ie end date was null but is updated to be non-null)

## ✅ Acceptance Criteria

- [ ] new computed gov employee fields added to user model
- [ ] gov_employee fields renamed in database with `computed_` prefix
- [ ] graphql User fields pointing to renamed columns continue to work as they currently do
- [ ] after user creates or updates a current work experience, or when updating a work experience previously marked as current, the computed gov employee fields are updated appropriately
- [ ] computeGovEmployeeProfile is very readable, maintainable, testable
- [ ] phpunit tests for the computeGovEmployeeProfile working how and when expected
- [ ] Saving a work experience can make a user a gov employee or not without rebooting app
  - [ ] experience save can make user stale

## 🛑 Blockers

```[tasklist]
### Blocked By
```

**Comment by tristan-orourke** [comment]
I'm open to other implementations... we're essentially copying values from one table to another right now. I considered just storing a pointer to the most current gov work experience, or calculating it in a relationship. But I think doing it like this makes some other things easier, like filtering by gov fields, or storing fields in snapshot. Maybe?

I've been staring at this ticket too long 😵

**Comment by tristan-orourke** [comment]
Do we have to let urql know that updating a work experience can make User stale?

**Comment by tristan-orourke** [comment]
> Question: What if the employee has two govt positions starting at the same time which experience we would go with?

This is already covered in the pseudocode, but in short, use the one with the most recent start date

```
else if **government experience** exists and at least one is current
  find the **current** government experience with the **most recent start date**
```

**Comment by vd1992** [comment]
There is no field called `gov_expected_end_date`

**Comment by gobyrne** [comment]
I guess we will need an additional column called "current government employment type" in our tables (besides Group-level) that has one of these values:

1. Casual
2. Term
3. Indeterminate (substantive)
4. Acting
5. Secondment
6. Assignment

I'm not sure if that overlaps with these:
> Create new fields in the database, on the users table.
> 
> computed_is_gov_employee - defaults to match is_gov_employee
> computed_gov_employee_type - defaults to match gov_employee_type

**Comment by petertgiles** [comment]
Discussion from refinement: leave computed employment type columns separate to mirror storage on experience model

**Comment by petertgiles** [comment]
@gobyrne I think what you're looking for is covered by computed_gov_employee_type (casual, term, indeterminate) and computed_gov_position_type (acting, secondment, assignment).

**Comment by petertgiles** [comment]
Replaced by:
- https://github.com/GCTC-NTGC/gc-digital-talent/issues/12643
- https://github.com/GCTC-NTGC/gc-digital-talent/issues/12645
- https://github.com/GCTC-NTGC/gc-digital-talent/issues/12648
